### PR TITLE
+doc #17885 explain sender semantics in receiveRecover

### LINKED ISSUE #17885: Akka Persistence migration guide should explain sender is not available in recovery
The change https://github.com/akka/akka/pull/17744 means that we do not have the sender (from the serialized reference) available during recovery. There might be code which assumed that the sender is an actual sender (well it would be weird to rely on this, but for limited cases it perhaps even worked).

We should document that sender() in `receiveRecover` is now always `deadLetters`

Reported by @chbatey, thanks!

### PR DESCRIPTION
**Pull request by ktoso**
Resolves #17885

// cc @chbatey, thanks for reporting again!

### CODE REVIEW

**Comment by chbatey** [comment]
Quick turnaround! Worth mentioning in migration-guide-persistence-experimental-2.3.x-2.4.x.rst as well in case people accidentally relied on the old behaviour? Btw it was really @krasserm's awesome tests that found it :)

**Comment by ktoso** [comment]
True, I'll add a mention to the guide as well. The old behaviour wasn't documented so relying on it was relying on undefined behaviour but yeah, Marin knows that codebase :-) I'll amend the commit.

**Comment by patriknw** [comment]
LGTM

**Comment by drewhk** [comment]
LGTM

**Comment by ktoso** [comment]
Added a word or two in the migration guide, in case someone relied on this undocumented behaviour.
Give it a quick look, I'll merge tomorrow

File: akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst [hunk 1]
```diff
@@ -353,6 +353,19 @@ as reference::
       // ...
     }
 
+Sender reference of replayed events is deadLetters
+--------------------------------------------------
+While undocumented, previously the ``sender()`` of the replayed messages would be the same sender that originally had
+sent the message. Since sender is an ``ActorRef`` and those events are often replayed in different incarnations of
+actor systema and during the entire lifetime of the app, relying on the existence of this reference is most likely
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst]: typo systema

File: akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst [hunk 2]
```diff
@@ -353,6 +353,19 @@ as reference::
...
+not going to succeed. In order to avoid bugs in the style of "it worked last week", the ``sender()`` reference is now not
+stored, in order to avoid potential bugs which this could have provoked.
+
+The previous behaviour was never documented explicitly (nor was it a design goal), so it is unlikely that applications
+have explicitly relied on this behaviour, however if you find yourself with an application that did exploit this you
+should rewrite it to explicitly store the ``ActorPath`` of where such replies during replay may have to be sent to,
+instead of relying on the sender reference during replay.
+
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst]: good!

**Comment by patriknw** [comment]
LGTM, after pr validation and fixing the typo

File: akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst]: el systemo, giving it a more spanish twist :wink:  // fixing

**Pull request merged**